ExampleProtocols.thy

```diff
@@ -61,4 +61,39 @@ lemma (in Protocol) latest_estimates_type :
 lemma (in Protocol) non_equivocating_validators_have_at_most_one_latest_message:
   "\<forall> v \<in> V. v \<notin> equivocating_validators \<sigma> \<longrightarrow> card (latest_message \<sigma> v) \<le> 1"
   oops
+
+(* Definition 4.10 *)
+(* TODO *)
+
+(* Lemma 6 *)
+(* TODO *)
+
+(* Lemma 7 *)
+(* TODO *)
+
+(* Lemma 8 *)
+(* TODO *)
+
+(* Lemma 9 *)
+(* TODO *)
+
+(* Lemma 10 *)
+(* TODO *)
+
+(* Definition 4.11: Latest messages from non-Equivocating validators *)
+definition  (in Protocol) latest_messages_from_non_equivocating_validators :: "state \<Rightarrow> validator \<Rightarrow> message set"
+  where
+    "latest_messages_from_non_equivocating_validators \<sigma> v = (if v \<in> equivocating_validators \<sigma> then \<emptyset> else latest_message \<sigma> v)"
+
+(* Definition 4.12: Latest honest message driven estimator *)
+(* TODO *)
+
+(* Definition 4.13: Latest estimates from non-Equivocating validators *)
+definition (in Protocol) latest_estimates_from_non_equivocating_validators :: "state \<Rightarrow> validator \<Rightarrow> consensus_value set"
+  where
+    "latest_estimates_from_non_equivocating_validators \<sigma> v = {est m | m. m \<in> latest_messages_from_non_equivocating_validators \<sigma> v}"
```

Comment:
I'm still not sure which is better but I explicitly limit `m` to be in `M` in other places. e.g.
```
{est m | m. m ∈ M ∧ m ∈ latest_messages_from_non_equivocating_validators σ v}
```

Reply:
See `DESIGN.md`.

Reply:
9d938b786e6db91d9f4a5c2d9912a1bc2f625fa3